public-ui/src/app/views/pm/pm.component.ts

```diff
@@ -1,15 +1,228 @@
+        maleCount: 0, femaleCount: 0, otherCount: 0},
+      {conceptId: '903115', conceptName: 'Diastolic', analyses: null, chartType: this.chartType,
+        maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]},
+    { group: 'height', groupName: 'Height', concepts: [
+      {conceptId: '903133', conceptName: 'Height', analyses: null, chartType: this.chartType,
+        maleCount: 0, femaleCount: 0, otherCount: 0}
+    ]},
+    { group: 'weight', groupName: 'Weight', concepts: [
+      {conceptId: '903121', conceptName: 'Weight', analyses: null, chartType: this.chartType,
+        maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]},
+    { group: 'mean-waist', groupName: 'Mean waist circumference', concepts: [
+      { conceptId: '903135', conceptName: 'Mean waist circumference', analyses: null,
+        chartType: this.chartType, maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]},
+    { group: 'mean-hip', groupName: 'Mean hip circumference', concepts: [
+      {conceptId: '903136', conceptName: 'Mean hip circumference', analyses: null,
+        chartType: this.chartType, maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]},
+    { group: 'mean-heart-rate', groupName: 'Mean heart rate', concepts: [
+      {conceptId: '903126', conceptName: 'Mean heart rate', analyses: null,
+        chartType: this.chartType, maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]},
+    { group: 'wheel-chair', groupName: 'Wheel chair use', concepts: [
+      {conceptId: '903111', conceptName: 'Wheel chair use', analyses: null, chartType: 'column',
+        maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]},
+    { group: 'pregnancy', groupName: 'Pregnancy', concepts: [
+      {conceptId: '903120', conceptName: 'Pregnancy', analyses: null, chartType: 'column',
+       maleCount: 0, femaleCount: 0, otherCount: 0},
+    ]}
+  ];
+
+  // Initialize to first group and concept, adjust order in groups array above
+  selectedGroup = this.conceptGroups[0];
+  selectedConcept = this.selectedGroup.concepts[0];
+
+  // we save the total gender counts
+  femaleCount = 0;
+  maleCount = 0;
+  otherCount = 0;
+
+  constructor(private api: DataBrowserService, public dbc: DbConstantsService) { }
+
+  loading() {
+    return this.loadingStack.length > 0;
+  }
 
   ngOnInit() {
+    this.showMeasurement(this.selectedGroup, this.selectedConcept);
+    // Get demographic totals
+    this.loadingStack.push(true);
+    this.subscriptions.push(this.api.getGenderAnalysis()
+      .subscribe(result => {
+        this.genderAnalysis = result;
+        for (const g of this.genderAnalysis.results) {
+          if (g.stratum1 === this.dbc.FEMALE_GENDER_ID) {
+            this.femaleCount = g.countValue;
+          } else if (g.stratum1 === this.dbc.MALE_GENDER_ID) {
+            this.maleCount = g.countValue;
+          } else {
+            this.otherCount += g.countValue;
+          }
+        }
+        this.loadingStack.pop();
```

Comment:
This is a nice pattern - should you be popping the stack if the request fails as well?

Reply:
Good catch. Yes. I will do that.